src/backend-api/Controllers/DetailController.cs

```diff
@@ -500,5 +526,248 @@ private IActionResult GetProgramDetail(int id)
+       * Function returns the program detail information.
+       * Returns : {
+       *    "departmentName": String,
+       *    "totalCostOfActHardwareInDep": decimal,
+       *    "totalCostOfProgramsInDep": decimal,
+       *    "picture: partial URL (as string),
+       *    "countempsIDsInDep": int,
+       *    "jsonHardware": {
+       *        DefaultHardware [
+       *        ] list with all the default hardware
+       *     },
+       *     "jsonSoftware": {
+       *        DefaultSoftware [
+       *        ] list with all the default software
+       *     },
+       *     "listOfEmployees": [
+       *        {
+       *             "employeeName": string,
+       *             "hireDate": date,
+       *             "hardwareCostForEmp": decimal,
+       *             "programCostForEmp": decimal
+       *         },
+       *     ], For all the employees in this department
+       *      "listOfTablePrograms": [
+       *         {
+       *             "programName": string,
+       *             "programCount": int,
+       *             "programCostPerYear": decimal,
+       *             "programIsCostPerYear": bool
+       *         }, 
+       *     ], For all the programs that owned by this department that are not licenses,
+       *     "licensesList": [
+       *         {
+       *         "progName": string,
+       *         "countOfThatLicense": int
+       *         },
+       *     ] For all the programs that owned by this department that are licenses
+       * }
+       */
+        private IActionResult GetDepartmentDetail(int DepId)
+        {
+
+            //finding the department
+            var dep = _context.Department.Find(DepId);
+            // checking if the department actually exists and isn't deleted
+            if (dep == null || dep.IsDeleted == true)
+            {
+                return NotFound();
+            }
+            // if the department does exist...
+            else
+            {
+                // storing the partial picture url
+                string picture = $"/images/department/{DepId}";
+
+                //Cost of Programs per department value
+                decimal? TotalCostOfProgramsInDep = 0;
+
+                // cost of hardware per department value
+                decimal? TotalCostOfActHardwareInDep = 0;
+
+                // lambda to get the ids of the all the employees in the current department
+                var empsIDsInDep = _context.Employee.Where(x => x.DepartmentId == DepId && x.IsDeleted == false).Select(x => x.EmployeeId).ToList();
+
+                // lambda to count the number of employees in the current department
+                var CountEmpsInDep = _context.Employee.Where(x => x.DepartmentId == DepId && x.IsDeleted == false).Count();
+
+                // lambda to collect all the employees in the current department into a list
+                var empsInDep = _context.Employee.Where(x => x.DepartmentId == DepId && x.IsDeleted == false);
```

Comment:
Couldn't you start with `empsInDep`. And then do `countEmpsInDep = empsInDep.count()` and `empsIDsInDep = empsInDep.Select(x => x.EmployeeId).ToList()`